src/overlay/SurveyDataManager.h

```diff
@@ -0,0 +1,206 @@
+#pragma once
+
+// Copyright 2024 Stellar Development Foundation and contributors. Licensed
+// under the Apache License, Version 2.0. See the COPYING file at the root
+// of this distribution or at http://www.apache.org/licenses/LICENSE-2.0
+
+#include "main/Config.h"
+#include "overlay/Peer.h"
+#include "util/NonCopyable.h"
+#include "util/Timer.h"
+#include "util/UnorderedMap.h"
+
+#include "medida/meter.h"
+#include "xdr/Stellar-overlay.h"
+#include "xdr/Stellar-types.h"
+
+#include <chrono>
+#include <mutex>
+#include <optional>
+
+namespace stellar
+{
+
+enum class SurveyPhase
+{
+    // Survey is currently recording data
+    RECORDING,
+    // Recording complete. Survey data is available for reporting.
+    REPORTING,
+    // No active survey in progress. No data is being recording or reported.
+    INACTIVE
+};
+
+struct RecordingNodeData
+{
+    RecordingNodeData(uint64_t initialLostSyncCount);
+
+    // Peer change data
+    uint32_t mAddedAuthenticatedPeers = 0;
+    uint32_t mDroppedAuthenticatedPeers = 0;
+
+    // SCP stats
+    std::vector<std::chrono::nanoseconds> mSCPFirstToSelfLatencies;
```

Comment:
Duration and rate units make sense, yeah. I looked a bit into how medida's timers are implemented, and noticed that we default to CKMS algorithm for sampling, and report _previous_ window instead of the current one. I think this means we won't get meaningful values until `COLLECTING_PHASE_MAX_DURATION` is reached, which is unlikely under normal circumstances.

I think we shouldn't use CKMS here: from what I recall, it struggles with small samples sizes anyway, and in this case we get a sample every ~5 seconds for about ~30 minutes, so it's only a few hundred samples. I think we can use `SampleType::kSliding` instead; it should be able to handle slow sample growth well. It's a bit annoying that Timer doesn't allow manual sample type adjustment, but the underlying histogram does, so we can use that instead.

Reply:
Ah, that makes sense. Thanks for digging into this. I replaced the `Timer`s with `Histogram`s using `kSliding` in `b658c335cd8d16f5224d3457c7123d90c9836d12`